src/shims/windows/foreign_items.rs

```diff
@@ -340,14 +389,48 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // std-only shim.
                 this.write_scalar(Scalar::from_machine_isize(which.into(), this), dest)?;
             }
+            "CloseHandle" => {
+                let [handle] =
+                    this.check_shim(abi, Abi::System { unwind: false }, link_name, args)?;
+
+                match Handle::from_scalar(this.read_scalar(handle)?.check_init()?, this)? {
+                    Some(Handle::Thread(thread)) => this.detach_thread(thread)?,
+                    _ => throw_ub_format!("invalid handle"),
```

Comment:
Is it truly UB to call `CloseHandle` with a wrong handle?

On Linux, the equivalent (`close` on a bad FD) is certainly not UB.

Reply:
Docs say that it throws an exception under a debugger with an invalid handle, so I think doing something similar to that is the correct thing to do.

Reply:
See also [strict handle checking](https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-process_mitigation_strict_handle_check_policy) which raises an exception on use of an invalid handle.

Reply:
Is it legal for the implementation to do the equivalent of `rm -rf /` when the handle is invalid?

If not, then `throw_ub` is wrong. `throw_unsup_format` or `throw_machine_stop!(TerminationInfo::Abort(msg))` are probably more appropriate.